rsocket-core/src/main/java/io/rsocket/resume/InMemoryResumableFramesStore.java

```diff
@@ -298,30 +412,357 @@ public void request(long n) {}
   @Override
   public void cancel() {
     pauseImplied();
-    state = 0;
+    markDisconnected();
   }
 
   @Override
   public void subscribe(CoreSubscriber<? super ByteBuf> actual) {
-    final int state = this.state;
-    if (state != 2) {
-      resumeImplied();
-      logger.debug(
-          "Tag: {}. Subscribed at Position[{}] and ImpliedPosition[{}]",
-          tag,
-          position,
-          impliedPosition);
-      actual.onSubscribe(this);
-      synchronized (this) {
-        for (final ByteBuf frame : cachedFrames) {
-          actual.onNext(frame.retainedSlice());
-        }
+    //    logger.debug(
+    //        "Tag: {}. Subscribed at Position[{}] and ImpliedPosition[{}]",
+    //        tag,
+    //        firstAvailableFramePosition,
+    //        impliedPosition);
```

Comment:
Is this meant to be commented in, or perhaps done at trace?

Reply:
improved logging